template/src/lib/sdk/widgets/IsochroneWidget.js

```diff
@@ -0,0 +1,174 @@
+      dispatch(removeLayer('isolineLayer'));
+    };
+  }, [dispatch]);
+
+  useEffect(() => {
+    const handleLaunchIsochrone = async () => {
+      try {
+        const isochrone = await launchIsochrone(credentials, {
+          geom: latLong,
+          mode: selectedMode,
+          range: selectedRange,
+        });
+        updateIsochrone(isochrone);
+      } catch (error) {
+        dispatch(setError(`Isochrone error: ${error.message}`));
+      }
+    };
+
+    if (openIsochroneConfig && selectedMode && selectedRange) {
+      handleLaunchIsochrone();
+    }
+  }, [
+    dispatch,
+    updateIsochrone,
+    credentials,
+    openIsochroneConfig,
+    selectedMode,
+    selectedRange,
+    latLong,
+  ]);
+
+  if (credentials.apiKey === 'default_public') {
+    console.error(
+      'IsochroneWidget not visible, you need to provide a valid API KEY or login with OAuth'
+    );
+    return null;
+  }
+
+  return (
+    <Grid container direction='column'>
+      {!openIsochroneConfig ? (
+        <Button onClick={clickLaunchHandle} variant='outlined' color='primary'>
+          Launch isochrone
+        </Button>
+      ) : (
+        <Grid>
+          <Typography variant='subtitle2'>Isochrone</Typography>
+          <Divider className={classes.divider} />
+          <Button
+            className={classes.delete}
+            onClick={clickLaunchHandle}
+            variant='text'
+            color='primary'
+          >
+            Delete
+          </Button>
+          <Grid container direction='row' wrap='nowrap'>
+            <FormControl className={classes.formControl} size='small'>
+              <InputLabel id='age-native-simple-label'>Mode</InputLabel>
+              <Select
+                labelId='age-native-simple-label'
+                value={selectedMode}
+                onChange={handleChangeMode}
+                variant='outlined'
+              >
+                {Object.values(MODES).map((mode) => {
+                  return (
+                    <MenuItem key={mode} value={mode}>
+                      {mode}
+                    </MenuItem>
+                  );
+                })}
+              </Select>
+            </FormControl>
+            <FormControl className={classes.formControl} size='small'>
+              <InputLabel id='age-native-simple-label'>Distance</InputLabel>
+              <Select
+                labelId='age-native-simple-label'
+                value={selectedRange}
+                onChange={handleChangeRange}
+                variant='outlined'
+              >
+                {Object.entries(RANGES).map(([key, range]) => {
+                  return (
+                    <MenuItem key={key} value={range}>
+                      {range} min.
+                    </MenuItem>
+                  );
+                })}
+              </Select>
+            </FormControl>
+          </Grid>
+        </Grid>
+      )}
+    </Grid>
+  );
+}
+
+const useStyles = makeStyles((theme) => ({
```

Comment:
I don't dislike this position, but we should probably be consistent and move this to the top as in other modules... what do you think?

Reply:
I've just moved but I think that we should move to bottom in the future. I think that is more "friendly"